### PR TITLE
Branch disjoint set

### PR DESCRIPTION
**Pull request by 4ndrelim**
Anyone pls help vet for correctness and clarity! (for DS portion can alr)

### CODE REVIEW

File: src/test/java/dataStructures/disjointSet/weightedUnion/DisjointSetTest.java
Comment (euchangxian) [inline src/test/java/dataStructures/disjointSet/weightedUnion/DisjointSetTest.java:1]: tests are well-constructed and easy to understand! 👍

File: src/test/java/dataStructures/disjointSet/quickFind/DisjointSetTest.java
Comment (euchangxian) [inline src/test/java/dataStructures/disjointSet/quickFind/DisjointSetTest.java:1]: 👍

File: src/main/java/dataStructures/disjointSet/weightedUnion/README.md
Comment (euchangxian) [inline src/main/java/dataStructures/disjointSet/weightedUnion/README.md:43]: nice

**Review by euchangxian — APPROVED** [review]
LGTM 👍 
I believe a possible enhancement to the READMEs would be including the practical applications of the data structures implemented. This would allow for better understanding of the problems that the DS solves!

**Comment by 4ndrelim** [comment]
do not merge yet - will add in applications before merging

**Pull request merged**
